connectors/src/api/webhooks/webhook_slack.ts

```diff
@@ -427,6 +512,19 @@ const _webhookSlackAPIHandler = async (
   }
 };
 
+function isChannelNameWhitelisted(
+  remoteChannelName: string,
+  whiteListedChannelPatterns?: string[]
```

Comment:
Let's don `string`. It's a regex so we have full flexibility and it will greatly simplify the PR